### ISSUE TITLE
Structural pattern matching ignores zero-valued flags of enum.Flag with strict mode

### ISSUE THREAD

**Issue by schana** [microsoft/pyright#7576]
**Describe the bug**
When exhausting all the values within an `enum`, the wildcard match reports `error: Pattern will never be matched for subject type "Never"`, even though the zero-valued flag could still be matched.

**Code or Screenshots**
```python
from enum import Flag, auto

class Color(Flag):
    RED = auto()
    BLUE = auto()

value = Color(0)

match value:
    case Color.RED:
        print("red")
    case Color.BLUE:
        print("blue")
    case _:
        print("none")
```

**VS Code extension or command-line**

command-line
`pyright` version `1.1.356`

**Comment by erictraut** [comment]
Thanks for the bug report. I didn't realize that subclasses of `enum.Flag` allowed this behavior.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.357, which I just published.
